R/fftrees_wordstofftrees.R

```diff
@@ -44,42 +45,61 @@ fftrees_wordstofftrees <- function(x,
   )
 
 
-  # Clean up my.tree: ----
+  # Clean up and check my.tree: ------
 
-  # Split into one sentence:
+  # Collapse into one sentence:
   if (length(my.tree) > 1) {
     my.tree <- paste(my.tree, collapse = ". ")
   }
 
-  # Remove \n (can happen if my.tree has line breaks):
+
+  # Remove line breaks (\n):
   my.tree <- gsub(pattern = "\n", replacement = "", x = my.tree)
 
-  if (all(grepl(x$params$decision.labels[1], x = my.tree) == FALSE)) {
-    stop("Some decision.labels are not in my.tree.")
-  }
 
-  cue.names.l <- tolower(x$cue_names)
-  my.tree <- tolower(my.tree)
+  # Use lowercase spelling (for robustness against typos):
+  my.tree         <- tolower(my.tree)
+  cue.names.l     <- tolower(x$cue_names)
   decision.labels <- tolower(x$params$decision.labels)
 
 
-  # Split my.tree into def parts: ----
+  # Verify that both decision labels/exit types occur (at least once) in my.tree:
```

Comment:
I wonder why this is necessary. Is it to check to ensure that the supplied version of `my.tree` is valid? 

If yes, could it also be implemented as a helper function `assert_valid_my_tree()`? I'm ok to leave as is, I'm just wondering

Reply:
It's an improvement of the previous check, which merely looked for "False" decision labels.  Making sure that some "True" exit is specified somewhere in the description seemed a reasonable extension.  However, I also realized that _rake_ FFTs only need to specify a single exit type (as the "otherwise" part of the final node is being ignored). Thus, I now changed both `stop()`s into corresponding `warning()`s.  This warns the user, but still tries to do something reasonable with the definition — which seems reasonable here.

Incidentally, this allows for the verbal definition a new kind of "minimal" FFT:

```{r ex}
fft <- FFTrees(formula = survived ~., 
               data = titanic, 
               my.tree = "If sex = {female} true.")
```

which essentially assesses the performance of single-cue trees and could be really helpful when manually constructing FFTs.

While checking for _both_ types of decision labels in the `my.tree` description seems legitimate, it still falls short of a full test for the description's validity (which would, e.g., need to verify that _each_ conditional contained a valid decision label).  I fully agree that such a test would be useful and should use a separate helper function, but would prioritize addressing other issues first.

Reply:
Thanks! Very interesting discussion! All good for now.

Reply:
Yep — and I'll leave more comprehensive validity checking of user input for a future revision, although I was tempted to start immediately, as I am quite fond of the "verbal" `my.tree` option.  (I think it's one of the killer features of **FFTrees**, perhaps only rivaled by its plotting prowess.)